frontend/src/App.js

```diff
@@ -59,173 +60,187 @@ const NotFound = () => (
-                render={() => <ParticipantsPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/customer-review-questions"
-                render={() => <CustomerReviewQuestionsPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/peer-review-questions"
-                render={() => <PeerReviewQuestionsPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/registration-questions"
-                render={() => <RegistrationQuestionsPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/groups"
-                render={() => <GroupManagementPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/email-templates"
-                render={() => <EmailTemplatesPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/registrations"
-                render={() => <Registrations />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/reviews"
-                render={() => <InstructorReviews />}
-              />
-              <Route
-                exact
-                path="/customer-review/:id"
-                render={(props) => <CustomerReviewPage {...props} />}
-              />
-              <LoginRoute
-                exact
-                path="/register"
-                user={this.props.user}
-                render={() => <RegistrationPage />}
-              />
-              <LoginRoute
-                exact
-                path="/peerreview"
-                user={this.props.user}
-                render={() => <PeerReviewPage />}
-              />
-              <AdminRoute
-                exact
-                path="/administration/registrationmanagement"
-                render={() => <RegistrationManagementPage />}
-              />
-              <InstructorRoute
-                exact
-                path="/instructorpage"
-                render={() => <InstructorPage />}
-              />
-              <InstructorRoute
-                exact
-                path="/instructorreviewpage"
-                render={() => <InstructorReviewPage />}
-              />
-              <InstructorRoute
-                path="/adminstration/customer-reviews"
-                render={() => <ViewCustomerReviewsPage />}
-              />
-
-              <LoginRoute
-                exact
-                path="/registrationdetails"
-                render={() => <RegistrationDetailsPage />}
-              />
-              <Route component={NotFound} />
-            </Switch>
-          </div>
-        </div>
-      </Router>
-    )
+  const ConfigurationLoadingWrapper = ({ children }) => {
+    fetchConfigurations()
```

Comment:
My point for this was that when certain route is triggered, the information would be refreshed in the state. Since multiple, but not all, components placed to different routes need this information, I didn't think that it was necessary to load that data to state once on application loading but "on demand" if my idea is more clear this way?

Reply:
I'm worried that fetching on every render could result in unnecessary API calls 🤔 Could `useEffect` be used with `window.location.href` in the dependency array, or something similar to that?

Reply:
Couldn't make it work:
- window.location{.href/pathname} & fetchConfigurations in useEffect with condition of certain routes
- fetchConfigurations in useEffect without route condition

Could make it work:
- once when whole application loads but when changing to different view (configurations, registration management..), it didn't run the function again.
- the way it was. I could make the wrapper as component if desired to use useEffect in that?

Reply:
One viable option seems to be to provide fetchConfigurations as prop to components that needs the updated data and use that function inside the useEffect in the component.

Reply:
Found out that there was already a usage example in `TopicListPage` with `configurationPageActions.fetchConfigurations` and implemented it to other pages too. 
Luckily I started questioning why I was doing it such backwards fashion :laughing: 

And additionally fixed reversed data reversing in `TopicListPage` component so now those all components are showing right order.
And they use `useEffect` to refresh the configuration data in global state. _IF_ this is something we want to alter so that it won't be fetched too frequently, we need to implement something different than `useEffect`.